packages/server/src/conversations/api.ts

```diff
@@ -35,18 +43,23 @@ export class ConversationApi {
 
   async list(req: ClientApiRequest, res: Response) {
     const userId = req.params.userId as uuid
-    const limit = +(req.query.limit || 20)
+    const limit = +(req.query.limit || DEFAULT_LIMIT)
+
+    const user = await this.users.get(userId)
+    if (!user || user.clientId !== req.client.id) {
```

Comment:
I know we already had this discussion, but can you remind me why if you do a request to list the conversations of a user that is linked with a different client than yours, you would get a 404 not found instead of a 403 Forbidden?

In my mind, if you try to access a resource that you don't have permission to. Be it an admin page or user information, it should return 403, not 404.

Let's say I am a not user with normal privileges on a given website. If I were to access the admin page, I expect the error code to be 403 and then to redirect me back to where I was.

Reply:
Messaging is multi tenant for performance purposes, but we divide the database in different sectors to manage ressource ownership. The clientIds split the database into multiple smaller virtual databases if you want. So it makes sense in that context that trying to access a resource that's located outside your "virtual database" would return not found, since in your point of view the item is not located in your database.

Reply:
Fair point! As long as we are consistent here, let's go that way! And also, this makes me think that we should document that as it isn't that obvious!